pkg/model/job.go

```diff
@@ -213,43 +217,14 @@ func (s *Spec) AllStorageSpecs() []*StorageSpec {
 		}
 	}
 
+	if s.EngineSpec.Type == EngineWasm {
+		if v, ok := s.EngineSpec.Spec[WasmEngineEntryModuleKey].(StorageSpec); ok {
+			storages = append(storages, &v)
+		}
+	}
 	return storages
 }
 
-// for VM style executors
-type JobSpecDocker struct {
-	// this should be pullable by docker
-	Image string `json:"Image,omitempty"`
-	// optionally override the default entrypoint
-	Entrypoint []string `json:"Entrypoint,omitempty"`
-	// a map of env to run the container with
-	EnvironmentVariables []string `json:"EnvironmentVariables,omitempty"`
-	// working directory inside the container
-	WorkingDirectory string `json:"WorkingDirectory,omitempty"`
-}
-
-// Describes a raw WASM job
-type JobSpecWasm struct {
-	// The module that contains the WASM code to start running.
-	EntryModule StorageSpec `json:"EntryModule,omitempty"`
-
-	// The name of the function in the EntryModule to call to run the job. For
-	// WASI jobs, this will always be `_start`, but jobs can choose to call
-	// other WASM functions instead. The EntryPoint must be a zero-parameter
-	// zero-result function.
-	EntryPoint string `json:"EntryPoint,omitempty"`
-
-	// The arguments supplied to the program (i.e. as ARGV).
-	Parameters []string `json:"Parameters,omitempty"`
-
-	// The variables available in the environment of the running program.
-	EnvironmentVariables map[string]string `json:"EnvironmentVariables,omitempty"`
-
-	// TODO #880: Other WASM modules whose exports will be available as imports
-	// to the EntryModule.
-	ImportModules []StorageSpec `json:"ImportModules,omitempty"`
-}
-
```

Comment:
These now live in `model/engine_spec_docker.go`, `model/engine_spec_wasm.go`, `executor/docker/spec` and `executor/was/spec` along with their helper methods. They are duplicated across these package to avoid cyclical dependencies and to satisfy the implicit server<-> client relationship of requester and compute nodes.